integration_tests/src/main/python/hash_aggregate_test.py

```diff
@@ -2539,3 +2540,23 @@ def run_and_capture_plan(spark, get_df_function):
     # two-stage aggregate should be 2x the number of logical aggregates
     assert aggregate_nodes == 2, "Unexpected SparkPlan: " + plan_str
     assert_gpu_and_cpu_are_equal_collect(run_spark_fn, conf=run_conf)
+
+
+@pytest.mark.parametrize('data_gen', integral_gens, ids=idfn)
+def test_hash_reduction_bitwise(data_gen):
+    assert_gpu_and_cpu_are_equal_collect(
+        lambda spark: unary_op_df(spark, data_gen).selectExpr(
+            "bit_and(a)",
+            "bit_or(a)",
+            "bit_xor(a)"))
+
+
+@ignore_order(local=True)
+@pytest.mark.parametrize('int_gen', integral_gens, ids=idfn)
+def test_hash_groupby_bitwise(int_gen):
+    data_gen = [('a', IntegerGen()), ('b', int_gen)]
+    assert_gpu_and_cpu_are_equal_collect(
+        lambda spark: gen_df(spark, data_gen).groupby('a')
```

Comment:
I have the opposite problem with this test. The probability that we will get two a values that are the same in all 2048 values generated is very small. I think using a byte gen is probably better, but then we would run into the other problem that we would have, on average 8 entries per grouping. It would be more ideal if we had 1024 rows with a byte gen for the key. That would give us 4 values on average, which would give us different values each time.